src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerMetricsDiagnosticListener.cs

```diff
@@ -49,15 +49,33 @@ public override void OnEventWritten(string name, object payload)
                 {
                     var request = response.RequestMessage;
 
+                    TagList tags;
+
                     // TODO: This is just a minimal set of attributes. See the spec for additional attributes:
                     // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/http-metrics.md#http-client
-                    var tags = new KeyValuePair<string, object>[]
+                    if (!request.RequestUri.IsDefaultPort)
+                    {
+                        tags = new TagList
+                        {
+                            new KeyValuePair<string, object>(SemanticConventions.AttributeHttpMethod, HttpTagHelper.GetNameForHttpMethod(request.Method)),
+                            new KeyValuePair<string, object>(SemanticConventions.AttributeHttpScheme, request.RequestUri.Scheme),
+                            new KeyValuePair<string, object>(SemanticConventions.AttributeHttpStatusCode, (int)response.StatusCode),
+                            new KeyValuePair<string, object>(SemanticConventions.AttributeHttpFlavor, HttpTagHelper.GetFlavorTagValueFromProtocolVersion(request.Version)),
+                            new KeyValuePair<string, object>(SemanticConventions.AttributeNetPeerName, request.RequestUri.Host),
+                            new KeyValuePair<string, object>(SemanticConventions.AttributeNetPeerPort, request.RequestUri.Port),
```

Comment:
from the spec: `Conditionally Required: If not default (80 for http, 443 for https).`

IMO worth to check if it is really needed to add this attribute.

Reply:
this is already doing that right? port is only added if non-default.

Reply:
@Kielek - I am checking if it is not a default port first. do you mean the port should not be added at all?

Reply:
Sure... I have overlooked condition. Sorry!